crates/cxx-qt-lib/src/gui/qimage.cpp

```diff
@@ -20,15 +20,18 @@
 // 3. QImageData *d;
 // For a total of 3 pointers in length.
 // Because of the added v-table, it's a total of 4 pointers in size.
-assert_alignment_and_size(QImage,
-                          alignof(::std::size_t),
-                          sizeof(::std::size_t) * 4);
+constexpr static ::std::array<::std::size_t, 4> arr{ sizeof(ushort),
+                                                     sizeof(::std::size_t),
+                                                     sizeof(::std::size_t),
+                                                     sizeof(::std::size_t) };
+assert_alignment_and_size(QImage, alignof(::std::size_t), arr, arr.size());
 #else
 // In Qt6 the QPaintDevice doesn't contain the `reserved` pointer, making it 1
 // pointer smaller
-assert_alignment_and_size(QImage,
-                          alignof(::std::size_t),
-                          sizeof(::std::size_t) * 3);
+constexpr static ::std::array<::std::size_t, 3> arr{ sizeof(ushort),
+                                                     sizeof(::std::size_t),
+                                                     sizeof(::std::size_t) };
+assert_alignment_and_size(QImage, alignof(::std::size_t), arr, arr.size());
```

Comment:
Rust side uses either 3x or 4x of `usize` this should be changed to `u16` + 2/3x `usize`?

And then the C++ side should use `::std::uint16_t` instead of ushort as we want to use the modern numerics that enforce the type size to be the same as Rust?

Reply:
yes